### PR TITLE
Fix Secure Supply Chain compliance issues

### LINKED ISSUE #10374: Fix NPM compliance issues detected by Secure Supply Chain Analysis
### Problem Description

The Secure Supply Chain Analysis task runs on our publish and compliance pipelines, and has detected multiple issues with how we consume NPM packages.

Detection logic: https://docs.opensource.microsoft.com/tools/nuget_security_analysis/azure_artifacts_checks/
Specific remedies for each error type: https://aka.ms/cfs

```
##[warning]packages/e2e-test-app/.npmrc - CFS0002: missing registry option.
##[warning]packages/sample-apps/.npmrc - CFS0002: missing registry option.
##[warning]vnext/.npmrc - CFS0002: missing registry option.
##[warning]package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@office-iss/react-native-win32-tester/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@office-iss/react-native-win32/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/automation-channel/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/automation-commands/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/automation/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/cli/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/cli/src/e2etest/projects/BarPackage/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/cli/src/e2etest/projects/FooPackage/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/codegen/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/find-repo-root/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/fs/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/package-utils/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/telemetry/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/tester/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native-windows/virtualized-list/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native/repo-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@react-native/tester/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/babel-node-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/beachball-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/create-github-releases/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/doxysaurus/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/eslint-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/format-files/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/generated-beachball-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/integrate-rn/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/jest-debug-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/jest-e2e-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/jest-out-of-snapshot-resolver/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/jest-out-of-tree-resolver/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/jest-unittest-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/just-task/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/metro-dev-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/promote-release/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/stamp-version/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/take-screenshot/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/@rnw-scripts/ts-config/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/debug-test/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/integration-test-app/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/playground/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/react-native-platform-override/package.json - CFS0001: missing sibling .npmrc file.
##[warning]packages/react-native-windows-init/package.json - CFS0001: missing sibling .npmrc file.
##[warning]vnext/src/babel-plugin-codegen/package.json - CFS0001: missing sibling .npmrc file.
```

### Steps To Reproduce

See results from the task running in a pipeline: [Pipelines - Run RNW_0.0.0-canary.538 logs](https://dev.azure.com/microsoft/ReactNative/_build/results?buildId=53639076&view=logs&j=bd0c9a5d-cce5-5ddc-8479-a97ca35bf82d&t=552ee052-3a18-5501-c840-8c8a7e2fd1d9)

### Expected Results

_No response_

### CLI version

npx react-native --version

### Environment

```markdown
npx react-native info
```

### Target Platform Version

_No response_

### Target Device(s)

_No response_

### Visual Studio Version

_No response_

### Build Configuration

_No response_

### Snack, code example, screenshot, or link to a repository

_No response_

### LINKED ISSUE #11209: Secure Supply Chain Analysis warnings because of incorrect NuGet.config
### Problem Description

From [this compliance run](https://dev.azure.com/microsoft/ReactNative/_build/results?buildId=64338597&view=logs&j=27307724-e5a4-5138-9d40-a20492a83fb9&t=627bd5ed-c745-5927-bcf8-24960fc147f0):

```
> Starting Multifeed Nuget Security Analysis:
##[warning]packages/playground/windows/NuGet.Config - Multiple feeds declared.
> Starting Multifeed Corext Analysis:
> Starting Multifeed Python Security Analysis:
> Starting CFS NuGet Analysis:
##[warning]packages/playground/windows/NuGet.Config - CFS0013: package source has value that is not an Azure Artifacts feed.
##[warning]NuGet.Config - CFS0012: missing declared or inherited <clear /> element.
##[warning]packages/e2e-test-app/windows/NuGet.Config - CFS0012: missing declared or inherited <clear /> element.
##[warning]packages/integration-test-app/windows/NuGet.Config - CFS0012: missing declared or inherited <clear /> element.
##[warning]packages/sample-apps/windows/NuGet.Config - CFS0012: missing declared or inherited <clear /> element.
##[warning]vnext/NuGet.Config - CFS0012: missing declared or inherited <clear /> element.
```

### Steps To Reproduce

See compliance pipeline: https://dev.azure.com/microsoft/ReactNative/_build?definitionId=79546&_a=summary

### Expected Results

_No response_

### CLI version

npx react-native --version

### Environment

```markdown
npx react-native info
```

### Target Platform Version

10.0.19041

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

None

### Snack, code example, screenshot, or link to a repository

_No response_

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

Resolves Secure Supply Chain warnings.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why

To meet compliance requirements.

Closes #11209
Closes #10374

### What

* Removed invalid `nuget.config` configurations
* Removed unnecessary `.npmrc` files
* Temporarily left nuget.org feed in config (tracking: #11557)

## Screenshots
N/A

## Testing
N/A

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/11548)

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -4,5 +4,11 @@
     <add key="enabled" value="True" />
     <add key="automatic" value="True" />
   </packageRestore>
-  <!-- Note: Do not specify any NuGet feeds in this file, everything is available on the fallback NuGet.org. -->
+  <packageSources>
+    <clear />
+    <!-- Warning: Never add more NuGet feeds, to be compliant this repo must only rely on this single ADO feed. -->
+    <add key="react-native" value="https://pkgs.dev.azure.com/ms/react-native/_packaging/react-native-public/nuget/v3/index.json" />
+    <!-- Remove (comment out) the nuget.org feed once our ADO feed is configured correctly. See https://github.com/microsoft/react-native-windows/issues/11557 -->
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
```
Comment (dannyvv) [inline NuGet.Config:12]: Just FYI: Nuget fires requests to all feeds at the same time and first to return will be chosen. If you want ot populate our public feed you'll need ot make sure you have no nuget cache and remove this line...

File: .gitignore [hunk 2]
```diff
@@ -33,6 +33,12 @@ node_modules
 # Optional REPL history
 .node_repl_history
 
+# No NPM config to pass CFS compliance
+.npmrc
```
Comment (dannyvv) [inline .gitignore:37]: Confused about why we are ignoring lock files and .npmrc... Just like nuget.config we need to have a `.npmrc` at the root

**Comment by dannyvv** [comment]
Note: we'll need to do the same for `.npmrc` to be compliant.

**Review by dannyvv — APPROVED** [review]

File: NuGet.Config (hunk 1 above)
Comment (jonthysell) [reply NuGet.Config:12]: Yeah, the goal is once we're ready to turn on upstream for `react-native-public` then I'll remove the nuget.org line and going forward we'll be compliant.

File: .gitignore (hunk 2 above)
Comment (jonthysell) [reply .gitignore:37]: Not according to the task - I removed .npmrc and it passes.

**Pull request merged**